solution/ui/regulations/eregs-vite/src/components/policy-repository/SubjectSelector.vue

```diff
@@ -15,42 +20,140 @@ const props = defineProps({
 const $router = useRouter();
 const $route = useRoute();
 
+const state = reactive({
+    filter: "",
+    subjects: [],
+});
+
+watch(
+    () => props.policyDocSubjects.loading,
+    (loading) => {
+        if (loading) {
+            state.subjects = [];
+            return;
+        }
+
+        state.subjects = props.policyDocSubjects.results;
+    }
+);
+
+const getFilteredSubjects = (filter) => {
+    if (!filter || filter.length < 1) {
+        state.subjects = props.policyDocSubjects.results;
+        return;
+    }
+
+    state.subjects = props.policyDocSubjects.results.reduce((acc, subject) => {
+        const shortNameMatch = subject.short_name
+            ? subject.short_name.toLowerCase().includes(filter.toLowerCase())
+            : false;
+        const abbreviationMatch = subject.abbreviation
+            ? subject.abbreviation.toLowerCase().includes(filter.toLowerCase())
+            : false;
+        const fullNameMatch = subject.full_name
+            ? subject.full_name.toLowerCase().includes(filter.toLowerCase())
+            : false;
+
+        if (shortNameMatch || abbreviationMatch || fullNameMatch) {
+            let displayName;
+
+            if (shortNameMatch) {
+                displayName = subject.short_name;
+            } else if (abbreviationMatch) {
+                displayName = subject.abbreviation;
+            } else if (fullNameMatch) {
+                displayName = subject.full_name;
+            }
+
+            displayName =
+                "<span class='match__container'>" +
+                displayName.replace(
+                    new RegExp(filter, "gi"),
+                    (match) => `<span class="match">${match}</span>`
+                ) +
+                "</span>";
+
+            const newSubject = {
+                ...subject,
+                displayName,
+            };
+
+            return [...acc, newSubject];
+        }
+
+        return acc;
+    }, []);
+};
+
+const debouncedFilter = _debounce(getFilteredSubjects, 100);
+
+watch(() => state.filter, debouncedFilter);
+
 const subjectClick = (event) => {
     const subjects = $route?.query?.subjects ?? [];
     const subjectsArray = _isArray(subjects) ? subjects : [subjects];
-    const subjectToAdd = event.target.dataset.id;
+    const subjectToAdd = event.currentTarget.dataset.id;
```

Comment:
target vs currentTarget is one of those JS gotcha questions that trip up even seasons devs in job interviews